charts/alchemist/templates/deployment.yaml

```diff
@@ -47,7 +49,7 @@ spec:
             valueFrom:
               secretKeyRef:
                 name: {{ quote .Values.global.redis.passwordSecret }}
-                key: {{ quote .Values.global.redis.passwordSecret }}
+                key: {{ .Values.global.redis.passwordSecretKey | default .Values.global.redis.passwordSecret | quote }}
```

Comment:
Use the passwordSecretKey, or use the name of the secret title itself. Useful in the event that the name and key attributes of secretKeyRef are the same.